src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -1,6 +1,6 @@
 (ns puppetlabs.puppetdb.jdbc
```

Comment:
As mentioned in chat, the retry code put in to work around the BoneCP bug for not handling 57P01 can be removed now most probably: https://github.com/puppetlabs/puppetdb/blob/master/src/puppetlabs/puppetdb/jdbc.clj#L300-L305

By having it still there, my testing is under question now, since it might not have been doing the right thing in the pooler, but I believe I've tested a 57P01 case under HikariCP and it works just fine.